src/resolve/arithmetic.rs

```diff
@@ -159,22 +161,21 @@ impl Resolver for ArithmeticResolver {
 mod tests {
     use crate::ast::Term;
     use crate::query_dfs;
-    use crate::resolve::ResolverExt;
+    use crate::resolve::{OrElse, ResolverExt};
     use crate::search::Solution;
     use crate::textual::TextualUniverse;
 
     use super::ArithmeticResolver;
 
     #[test]
     fn simple() {
-        let mut tu = TextualUniverse::new();
-        let query = tu
+        let tu = TextualUniverse::new();
+        let mut query = tu
             .prepare_query("is(X, add(3, mul(3, sub(6, div(10, rem(10, pow(2,3))))))).")
             .unwrap();
-        let mut results = query_dfs(
-            ArithmeticResolver::new(&mut tu.symbols).or_else(tu.resolver()),
-            &query,
-        );
+        let resolver = ArithmeticResolver::new(&mut query.symbols_mut());
+        let resolver: OrElse<ArithmeticResolver, _> = resolver.or_else(tu.resolver());
```

Comment:
Was this type annotation necessary to make trait resolution work, or did you simply add it to make things more clear?

Reply:
it was just a debugging sanity check. I removed it.